Refactor turrets to use bitflags and fix some bugs

Irafas opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Noticed while playing AI that the upload turrets weren't firing at at a Head so started this to fix that. While working I realised that #46637 aimed to fix that and was merged. Still though I saw some issues in the code and I wanted to refactor the 9 shooting vars into bitflags. This should resolve some bugs and generally be a bit better.
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Bugs are bad and about a year ago @Qustinnus in #36226 said I should use bitflags, so I finally got around to it.

Explanation of changes:
- Converted turrets to use bitflags
- AI turrets are set to shoot heads by default via flags. Removed the fix from #46637 that had the turretid machine doing it
- The var "criminals" didn't actually do anything. The "check_records" var is what actually shot wanted criminals. Removed to cut flags to eight
- There was a check in process() that would run after a perp was assessed by assess_perp(). If the turret was ignoring heads and the perp was a head it would discard them. This was a bit redundant as there was a check inside of assess_perp() that discards heads.
- That same check in assess_perp() was also using get_id_name() which returns the name, not the job. Changed it to use get_assignment() instead
- The position of the check that would add borgs to the targets list was never being reached so borgs would never be shot even when the option was toggled. This is because of a check before that step that discards silicons inside the same faction as the turret, which for upload turrets is "silicon". I've moved that code above the faction check so borgs will be shot
- Clicking the button in the turretid that toggles borg shooting didn't work because it called the wrong operation. Fixed that

@MacBlaze1 Let me know if this breaks any of the intended behavior from your PR's. I did some testing and it seems to be working as was intended
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
fix: Turretid button to toggle borg shooting works
fix: Turrets now can shoot borgs when toggled to
refactor: Turrets use bitflags instead of several vars
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Irafas committed:
Fixes and refactor

-Refactors the turret to use flags instead of a large number of boolean variables
-The variables "check_records" and "criminals" had the same general function but "criminals" didn't actually do anything. So it has been essentially removed
-Removed a redundant check in process() for whether the turret should shoot a Head. This would only run the check if the target was assessed to be a threat and would then immediately discard the target. Instead the target is now discarded in assess_perp() which has also had a couple changes detailed in this commit.
-Adjusted the "shoot Heads" check in assess_perp() so that it actually checks the ID assignment, as before it was using the ID name.
-Adjusted the "shoot Heads" check in assess_perp() so that if its set to not shoot Heads it will always return 0 for a Head ID. This is opposed to before where it would only check if it is also shooting those without mindshields.
-AI turrets are set to shoot Heads

Irafas committed:
Merge branch 'master' into upload-turrets-merge

Irafas committed:
Complete merge and fix turretid

-Fix merge issues that called old variables
-Fixed bug where the "shoot cyborgs" button on turretid's didn't actually do anything
-The turretid's button text for switching shooting borgs was the wrong way around

Irafas committed:
Redundant cyborg check

-The check inside assess_perp() for whether the turret should fire upon a borg will never be reached. assess_perp() is only ever called with humans. There is already a separate check in process() for borgs, though that doesn't work due to factions.

Irafas committed:
Shifted borg attack check

-Because of the order of checks borgs would never be fired upon. This commit just shifts the check above the faction check, which is what was causing the prior issues.

Irafas committed:
Shifted Head shooting from turretid to turret

-Shifted assignment of TURRET_FLAG_SHOOT_BORGS to the turret ../porta_turret/ai type and away from the turretid thing

Irafas committed:
Grammar change

-Had the "Yes" and "No" the wrong way roun

Qustinnus commented on code/game/machinery/porta_turret/portable_turret.dm:
```diff
@@ -204,21 +204,19 @@
-				check_records = !check_records
-			if("shootcrooks")
-				criminals = !criminals
+				TOGGLE_BITFIELD(turret_flags, TURRET_FLAG_AUTH_WEAPONS)
```
these ugly ass defines still exist? smh

Irafas replied:
Should I switch the cases where I've used it to `turret_flags ^= FLAG` instead?
Did a quick check and there is only one other instance of TOGGLE_BITFIELD being used in the entire game as well.

Qustinnus replied:
Yeah, I'm removing those dumb defines in a PR

Qustinnus commented on code/game/machinery/porta_turret/portable_turret.dm:
```diff
@@ -4,6 +4,15 @@
 #define POPUP_ANIM_TIME 5
 #define POPDOWN_ANIM_TIME 5 //Be sure to change the icon animation at the same time or it'll look bad
 
+#define TURRET_FLAG_BEEN_ATTACKED		(1<<0)	// The turret gets pissed off and shoots at people nearby (unless they have sec access!)
```
I feel like this shouldn't be in this flag list and should stay a bool because it doesn't really fit in with the rest of the checks.

Irafas replied:
Yeah I wasn't too sure about that one while writing this. It basically functions like a temporary TURRET_FLAG_SHOOT_ALL so I decided to include it with the flags since it works like them.

AnturK replied:
This is fine but probably change name to something like SHOOT_ALL_REACTION or similar so it's obvious the effect is the same.

Irafas committed:
🦀UGLY DEFINES ARE GONE🦀

optimumtact commented:
based and flagpilled, now stop using emojii in your commit logs

Qustinnus commented:
also lol i only just realized i told you to make this into flags like a year ago

MacBlaze1 commented:
If it shoots heads for portable, set start upload ones and the same for cyborgs but all toggled on or off and working, then I fully support this. Turret code is a mess.

AnturK commented on code/game/machinery/porta_turret/portable_turret.dm:
```diff
@@ -306,10 +304,9 @@
-		criminals = pick(0, 1)
-		auth_weapons = pick(0, 1)
-		stun_all = pick(0, 0, 0, 0, 1)	//stun_all is a pretty big deal, so it's least likely to get turned on
+		turret_flags |= pick(TURRET_FLAG_SHOOT_CRIMINALS, 0)
```
If you're touching these just put these behind `if(prob(xx))`

AnturK commented on code/game/machinery/porta_turret/portable_turret.dm:
```diff
@@ -373,16 +370,17 @@
 			if(ispAI(A))
 				continue
 
+			if(iscyborg(sillycone) && (turret_flags & TURRET_FLAG_SHOOT_BORGS) && sillycone.stat != DEAD)
```
Move to flag check to the front so it doesn't do istype unless it has to.

AnturK reviewed (changes requested):
Just some minor cleanup and it's all good.

Irafas committed:
AnturK's requested cleanup

- Renamed TURRET_FLAG_BEEN_ATTACKED to TURRET_FLAG_SHOOT_ALL_REACT
- Using if(prob(X)) instead of pick(X,Y, ...) for EMP effects
- Improved an if by moving the most costly check to the back
